src/Workspaces/Core/Portable/FindSymbols/FindReferences/DependentTypeFinder_ProjectIndex.cs

```diff
@@ -13,18 +13,22 @@ namespace Microsoft.CodeAnalysis.FindSymbols
 {
     internal static partial class DependentTypeFinder
     {
-        private class ProjectIndex
+        private sealed class ProjectIndex
         {
-            private static readonly ConditionalWeakTable<Project, AsyncLazy<ProjectIndex>> s_projectToIndex =
-                new();
+            private static readonly ConditionalWeakTable<ProjectState, AsyncLazy<ProjectIndex>> s_projectToIndex = new();
```

Comment:
this is much nicer too.  If the project *state* (not instance) hasn't changed, we can use thsi entire textual index.  The project-state instance will not change if we change another project, which is the semantics we would prefer here.  Previously, any edit anywhere would cause all these indices for all proejcts to be dropped.